### PR TITLE
Args structs for vehicle game commands

### PR DESCRIPTION
**Pull request by duncanspumpkin**
Switching to using game command args structs for the game commands. This will make things easier to use when implementing game commands and brings things into more standardised structure. (also will make changing the game command flags to an enum much easier)

### CODE REVIEW

File: src/OpenLoco/src/GameCommands/GameCommands.h [hunk 1]
```diff
@@ -143,14 +143,28 @@ namespace OpenLoco::GameCommands
...
+        }
+
+        EntityId source;
+        EntityId dest;
+
+        explicit operator registers() const
+        {
+            registers regs;
+            regs.di = enumValue(source);
+            regs.dx = enumValue(dest);
```
Comment (duncanspumpkin) [inline src/OpenLoco/src/GameCommands/GameCommands.h:164]: accidentally swapped

**Comment by duncanspumpkin** [comment]
I'm not 100% with the vehicle rename args. Open to suggestions. We use the same args structure for all the renames so would be good to be consistent.

File: src/OpenLoco/src/Windows/Vehicle.cpp [hunk 2]
```diff
@@ -492,9 +492,14 @@ namespace OpenLoco::Ui::Windows::Vehicle
                     onChangeDirection(&self);
                     break;
                 case widx::passSignal:
+                {
+                    GameCommands::VehicleAirPlacementArgs args{};
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/Windows/Vehicle.cpp:496]: There's a mistake here, should be `VehiclePassSignalArgs`.

File: src/OpenLoco/src/GameCommands/GameCommands.h [hunk 3]
```diff
@@ -286,13 +286,25 @@ namespace OpenLoco::GameCommands
...
 
-    inline void do_6(EntityId car)
+    struct VehicleSellArgs
     {
-        registers regs;
-        regs.bl = Flags::apply;
-        regs.dx = enumValue(car);
-        doCommand(GameCommand::vehicleSell, regs);
-    }
+        static constexpr auto command = GameCommand::vehiclePassSignal;
```
Comment (AaronVanGeffen) [inline src/OpenLoco/src/GameCommands/GameCommands.h:291]: There's a mistake here. Should be the sell game command.

**Pull request merged**